internal/entities/utils/encoding.go

```diff
@@ -24,6 +26,26 @@ func NewBigFloat(value *big.Float) *BigFloat {
 	return (*BigFloat)(value)
 }
 
+func (bf *BigFloat) MarshalJSON() ([]byte, error) {
+	if bf == nil {
+		return []byte("null"), nil
+	}
+	valueStr := bf.Native().Text('f', -1)
+	return json.Marshal(valueStr)
```

Comment:
I think this will marshall as string, I think is better to marshall this field as a number as the range is quite low [0;100]. Also this is what is forcing you to implement a custom validator